compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -4216,70 +4202,58 @@ private BLangMarkdownDocumentation createMarkdownDocumentationAttachment(Optiona
-//    |   | |                        | |                           |
-//    |   \_/non-tick (un-quoted)    \_/non-tick (in double quote) |
-//    \____________________________________________________________/
-//
-// note this state machine is only design to distinguish single tick and double tick quoted string
-// for simplicity are assuming three tick case is just a subset of double tick case
-
-        int length = text.length();
-        int state = startsInsideQuotes ? 2 : 0;
-        int lastWhiteSpace = 0;
-        int secondToLastWhiteSpace = 0;
-
-        //                0  1  2  3  4
-        int[][] table = {{0, 4, 2, 2, 4},  // non-tick
-                         {1, 2, 3, 0, 0}}; // tick
-
-        for (int i = 0; i < length; i++) {
-            char c = text.charAt(i);
-            int isTick = c == '`' ? 1 : 0;
-            int newState = table[isTick][state];
-
-            if (newState == 0 && c == ' ') {
-                secondToLastWhiteSpace = lastWhiteSpace;
-                lastWhiteSpace = i;
-            }
+    private String addReferencesAndReturnDocumentationText(LinkedList<BLangMarkdownReferenceDocumentation> references,
+                                                           NodeList<Node> docElements) {
+        StringBuilder docText = new StringBuilder();
+        for (Node element : docElements) {
+            docText.append(element.toString());
 
-            if (state == 4 && newState == 0) { // we are coming out of a single quoted string
-                BLangMarkdownReferenceDocumentation ref = new BLangMarkdownReferenceDocumentation();
-                ref.type = DocumentationReferenceType.BACKTICK_CONTENT;
-                if (secondToLastWhiteSpace + 1 < lastWhiteSpace) {
-                    ref.type = stringToRefType(text.substring(secondToLastWhiteSpace + 1, lastWhiteSpace));
-                }
-                references.add(ref);
+            // Add references if available
+            if (element.kind() == SyntaxKind.DOCUMENTATION_REFERENCE) {
+                BLangMarkdownReferenceDocumentation bLangRefDoc = new BLangMarkdownReferenceDocumentation();
+                DocumentationReferenceNode docReferenceNode = (DocumentationReferenceNode) element;
+
+                Token backtickContent = docReferenceNode.backtickContent();
+                bLangRefDoc.referenceName = backtickContent.isMissing() ? "" : backtickContent.text();
+
+                bLangRefDoc.type = DocumentationReferenceType.BACKTICK_CONTENT;
+                docReferenceNode.referenceType().ifPresent(
+                        refType -> bLangRefDoc.type = stringToRefType(refType.text())
+                );
+
+                references.add(bLangRefDoc);
             }
-            state = newState;
         }
-        return state == 2 || state == 3;
+
+        String text = docText.toString();
+        if (text.endsWith("\n")) {
+            text = text.substring(0, text.length() - 1);
+        }
```

Comment:
Can you explain whats happening in tis snippet?

I think you'l have to consider windows environments as well, since in windows, theres a carriage return (\r) for newlines.

Reply:
To get the full string containing in the `NodeList<Node> docElements`. I loop through it and append all `docElement.toString()`s. In the last docElement there will be a `\n` in its last children's trailing minutiae which should not be included in the documentation line text. 

yeah few parser tests too are falling in windows build. That's should be due  to above reason. I'll fix

Reply:
I think the issue is using `element.toString()`. That can cause issues  since `toString()` includes minutiae such as newlines, invalid tokens etc. We should use `.text()` method in the Token class

Reply:
We will have a problem for lines having documentation reference. e.g  text of ``annotation `field a` documentation`` will be ``annotation`field a`documentation``   if we use `.text()`, as white spaces there are captured as trivia. Shall I use `toString()` anyway? and replace `\r\n` and  `\n` with `""` at the end

Reply:
ah ok. +1.
But please make sure it won't include other minutiae such as invalid nodes, comments, etc.